src/core_modules/capture-core/components/WidgetsChangelog/common/hooks/useChangelogData.js

```diff
@@ -67,54 +90,109 @@ export const useChangelogData = ({
         },
     );
 
-    const records: ?Array<ChangelogRecord> = useMemo(() => {
-        if (!data) return undefined;
-
-        return data.changeLogs.map((changelog) => {
-            const { change: apiChange, createdAt, createdBy } = changelog;
-            const elementKey = Object.keys(apiChange)[0];
-            const change = apiChange[elementKey];
-
-            const { metadataElement, fieldId } = getMetadataItemDefinition(
-                elementKey,
-                change,
-                dataItemDefinitions,
-            );
-
-            if (!metadataElement) {
-                log.error(errorCreator('Could not find metadata for element')({
-                    ...changelog,
-                }));
-                return null;
-            }
-
-            const { firstName, surname, username } = createdBy;
-            const { options } = metadataElement;
-
-            const previousValue = convert(
-                convertServerToClient(change.previousValue, metadataElement.type),
-                metadataElement.type,
-                options,
-            );
-
-            const currentValue = convert(
-                convertServerToClient(change.currentValue, metadataElement.type),
-                metadataElement.type,
-                options,
+    useEffect(() => {
```

Comment:
Can performance be improved? Right now, the `useEffect` code is executed every time the changeLog modal is opened. If there is no change to the data, we shouldn't compute the records again